### PR TITLE
Enhance delete method

### PR DESCRIPTION
**Pull request by nashousni**

### CODE REVIEW

File: afs/afs-core/src/main/java/com/powsybl/afs/Node.java [hunk 1]
```diff
@@ -26,8 +26,14 @@ protected Node(FileCreationContext context, int codeVersion, boolean folder) {
     }
 
     public void delete() {
-        storage.deleteNode(info.getId());
-        storage.flush();
+        if (isFolder()) {
+            Folder folderToDelete = (Folder) this;
+            if (folderToDelete.getChildren().isEmpty()) {
+                storage.deleteNode(info.getId());
+                storage.flush();
```
Comment (geofjamg) [inline afs/afs-core/src/main/java/com/powsybl/afs/Node.java]: flush should be done just one time when last folder is deleted

File: afs/afs-core/src/test/java/com/powsybl/afs/AfsBaseTest.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public void baseTest() {
         Folder dir41 = dir2.createFolder("dir41");
         Project project3 = dir41.createProject("project3");
         project3.delete();
-        assertTrue(dir41.getChildren().isEmpty());
+        assertTrue(!dir41.getChildren().isEmpty());
```
Comment (yichen88) [inline afs/afs-core/src/test/java/com/powsybl/afs/AfsBaseTest.java]: use assertFalse

File: afs/afs-core/src/main/java/com/powsybl/afs/Node.java (hunk 1 above)
Comment (nashousni) [reply afs/afs-core/src/main/java/com/powsybl/afs/Node.java]:
```  if (isFolder()) {
            Folder folderToDelete = (Folder) this;
            if (folderToDelete.getChildren().isEmpty()) {
                storage.deleteNode(info.getId());
            }
        }
           storage.flush();
```

Comment (nashousni) [reply afs/afs-core/src/main/java/com/powsybl/afs/Node.java]: which hierarchy? it only concerns non empty folders

**Pull request merged**